app/imports/models.coffee

```diff
@@ -149,21 +133,22 @@ Trigger = Parse.Object.extend 'Trigger',
   create: (props, form) ->
     @setProperties props
     @save()
-      .then (trigger) =>
+      .then =>
         @addToForm(form)
       .then ->
-        trigger.id
+        @id
```

Comment:
@jgoley `@id` is not in a fat arrow function here so it might have the wrong context.